validator-api/src/network_monitor/monitor/preparer.rs

```diff
@@ -167,10 +168,31 @@ impl PacketPreparer {
     }
 
     pub(crate) async fn wait_for_validator_cache_initial_values(&self) {
+        // wait for the cache to get initialised
         self.validator_cache.wait_for_initial_values().await;
+
+        // now wait for our "good" topology to be online
+        info!("Waiting for 'good' topology to be online");
+        let initialisation_backoff = Duration::from_secs(30);
+        loop {
+            let gateways = self.validator_cache.gateways().await;
+            let mixnodes = self.validator_cache.mixnodes().await;
+            if self
+                .tested_network
+                .is_online(&mixnodes.into_inner(), &gateways.into_inner())
+            {
+                break;
+            } else {
+                info!(
```

Comment:
30 seconds seems like quite a long time, maybe this should be a warning instead?

Reply:
It's only happening at startup and I'd even argue that perhaps 30s is too short. This is basically to help to solve our chicken and the egg problem of network monitor requiring "good" topology to be online to run and mixnodes/gateways requiring the validator-api to be online in order to start themselves. With this delay we can start the basic validator-api functionality (without monitoring), while we wait for mixnodes/gateways to become online. After that happens the monitor will start